pageserver/src/http/openapi_spec.yml

```diff
@@ -1314,10 +1314,11 @@ components:
     TenantLocationConfigRequest:
       type: object
       required:
-        - tenant_id
+        - mode
       properties:
         tenant_id:
           type: string
+          description: Only used for the storage controller.
```

Comment:
If you define "use" in the strict sense that there is non-error behaviour changes by specifying different values here, then the storage controller is not actually using it.

It does however return errors if the specified shard count is something other than 0, so if you define "use" in the sense of "where the value is read", then there is a use present.

I don't think this use is justifying the existence of the field though so IMO in the long term we should remove it entirely.

Reply:
Since the purpose of this PR is to remove usages to enable cleaning up later, let's remove the usages in this PR rather than leaving it as a todo.  If you just parse the `:tenant_id` in the request path as a TenantShardId, that gets you the value which can be asserted on, and you don't need to read this struct member any more.

Reply:
I've pushed a commit to make the shard ID part of the URL (and the tenant_id field never be read).